internal/controller/vulnerabilityreports_controller.go

```diff
@@ -0,0 +1,169 @@
+		}
+		if _, err := r.ConsoleClient.UpsertVulnerabilityReports([]*console.VulnerabilityReportAttributes{createAttribute(vulnerabilityReport, serviceID)}); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func createAttribute(vulnerabilityReport *trivy.VulnerabilityReport, serviceID string) *console.VulnerabilityReportAttributes {
+	var namespaces []*console.NamespaceVulnAttributes
+	var vulnerabilityAttributes []*console.VulnerabilityAttributes
+	os := &console.VulnOsAttributes{
+		Eosl:   lo.ToPtr(vulnerabilityReport.Report.OS.Eosl),
+		Family: lo.ToPtr(string(vulnerabilityReport.Report.OS.Family)),
+		Name:   lo.ToPtr(vulnerabilityReport.Report.OS.Name),
+	}
+	summary := &console.VulnSummaryAttributes{
+		CriticalCount: lo.ToPtr(int64(vulnerabilityReport.Report.Summary.CriticalCount)),
+		HighCount:     lo.ToPtr(int64(vulnerabilityReport.Report.Summary.HighCount)),
+		MediumCount:   lo.ToPtr(int64(vulnerabilityReport.Report.Summary.MediumCount)),
+		LowCount:      lo.ToPtr(int64(vulnerabilityReport.Report.Summary.LowCount)),
+		UnknownCount:  lo.ToPtr(int64(vulnerabilityReport.Report.Summary.UnknownCount)),
+		NoneCount:     lo.ToPtr(int64(vulnerabilityReport.Report.Summary.NoneCount)),
+	}
+	artifact := &console.VulnArtifactAttributes{
+		Registry:   lo.ToPtr(vulnerabilityReport.Report.Registry.Server),
+		Repository: lo.ToPtr(vulnerabilityReport.Report.Artifact.Repository),
+		Digest:     lo.ToPtr(vulnerabilityReport.Report.Artifact.Digest),
+		Tag:        lo.ToPtr(vulnerabilityReport.Report.Artifact.Tag),
+		Mime:       lo.ToPtr(vulnerabilityReport.Report.Artifact.MimeType),
+	}
+
+	services := []*console.ServiceVulnAttributes{
+		{
+			ServiceID: serviceID,
+		},
+	}
+	if vulnerabilityReport.Namespace != "" {
+		namespaces = []*console.NamespaceVulnAttributes{
+			{
+				Namespace: vulnerabilityReport.Namespace,
+			},
+		}
+	}
+
+	for _, v := range vulnerabilityReport.Report.Vulnerabilities {
+		vulnerabilityAttr := &console.VulnerabilityAttributes{
+			Resource:         lo.ToPtr(v.Resource),
+			FixedVersion:     lo.ToPtr(v.FixedVersion),
+			InstalledVersion: lo.ToPtr(v.InstalledVersion),
+			Severity:         lo.ToPtr(console.VulnSeverity(v.Severity)),
+			Score:            v.Score,
+			Title:            lo.ToPtr(v.Title),
+			Description:      lo.ToPtr(v.Description),
+			CvssSource:       lo.ToPtr(v.CVSSSource),
+			PrimaryLink:      lo.ToPtr(v.PrimaryLink),
+			Links:            algorithms.Map(v.Links, func(s string) *string { return &s }),
+			Target:           lo.ToPtr(v.Target),
+			Class:            lo.ToPtr(v.Class),
+			PackageType:      lo.ToPtr(v.PackageType),
+			PkgPath:          lo.ToPtr(v.PkgPath),
+			PublishedDate:    lo.ToPtr(v.PublishedDate),
+			LastModifiedDate: lo.ToPtr(v.LastModifiedDate),
+		}
+		vulnerabilityAttributes = append(vulnerabilityAttributes, vulnerabilityAttr)
+	}
+
+	return &console.VulnerabilityReportAttributes{
+		ArtifactURL:     nil,
```

Comment:
@michaeljguarino I'm not sure how to get this.

Reply:
you would get this by concatenating registry + repository + tag/digest, in the usual docker image url format.  Basically want it to be easily accessible for the ui, and also to potentially cross-ref elsewhere